### PR TITLE
Offline sync selector updates

### PR DESCRIPTION
**Pull request by szabinst**
Offline selector updates

refs: MBL-17104
affects: Student
release note: none
test plan: See ticket

## Screenshots

<table>
<tr><th>Dialog</th>
<tr>
<td></td>
</table>

## Checklist

- [ ] Follow-up e2e test ticket created
- [x] A11y checked
- [x] Tested on phone
- [ ] Tested on tablet
- [ ] Approve from product

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

File: Core/Core/CourseSync/CourseSyncSelector/ViewModel/CourseSyncSelectorViewModel.swift [hunk 1]
```diff
@@ -32,19 +32,27 @@ class CourseSyncSelectorViewModel: ObservableObject {
...
         title: String(localized: "Sync Offline Content?", bundle: .core),
         message: "", // Updated when selected item count changes
         cancelButtonTitle: String(localized: "Cancel", bundle: .core),
         confirmButtonTitle: String(localized: "Sync", bundle: .core),
         isDestructive: false
     )
 
+    public let cancelConfirmAlert = ConfirmationAlertViewModel(
+        title: String(localized: "Cancel Offline Content Sync?", bundle: .core),
+        message: "Selection changes that you may had made won't be saved. Are you sure you want to cancel?",
```
Comment (vargaat) [inline Core/Core/CourseSync/CourseSyncSelector/ViewModel/CourseSyncSelectorViewModel.swift:50]: Let's localize this line.

File: Core/Core/SwiftUIViews/ConfirmationAlert.swift
Comment (vargaat) [inline Core/Core/SwiftUIViews/ConfirmationAlert.swift:1]: Guess the changes in this file were accidental.

**Review by vargaat — CHANGES_REQUESTED** [review]
QA+1, I added some minor review findings.

File: Core/Core/CourseSync/CourseSyncSelector/ViewModel/CourseSyncSelectorViewModel.swift [hunk 2]
```diff
@@ -32,19 +32,33 @@ class CourseSyncSelectorViewModel: ObservableObject {
...
         isDestructive: false
     )
 
+    public let cancelConfirmAlert = ConfirmationAlertViewModel(
+        title: String(
+            localized: "Cancel Offline Content Sync?",
+            bundle: .core
+        ),
+        message: String(
+            localized: "Selection changes that you may had made won't be saved. Are you sure you want to cancel?",
```
Comment (rh12) [inline Core/Core/CourseSync/CourseSyncSelector/ViewModel/CourseSyncSelectorViewModel.swift:54]: This one is not added to xcstrings

Comment (szabinst) [reply Core/Core/CourseSync/CourseSyncSelector/ViewModel/CourseSyncSelectorViewModel.swift:54]: Pushed the missing change.

**Comment by rh12** [comment]
When everything is deselected and "Sync" is tapped we show the regular confirmation dialog, which states:
```
Sync Offline Content?
This will sync ~Zero KB content. It may result in additional charges from your data provider if you are not connected to a Wi-Fi network.
[ Cancel | Sync ]
```
As an improvement this could display something more relevant, like
```
Clear Offline Content?
No content was selected. This will clear any previously synced content from your device.
[ Cancel | Clear ]
```
(A related thought to change the blue "Sync" button to "Clear local content" on the sync screen)

This is just and idea and may be out of scope in this PR.

**Review by rh12 — APPROVED** [review]
QA + 1
Minor thought: would it make sense to only show the cancel confirmation dialog if there are actually any changes in the selection?
If so, the confirmation message could be simplified a bit, as well.

**Comment by szabinst** [comment]
@rh12 I thought about it but that would increase the scope for this rather simple modification.

**Review by vargaat — APPROVED** [review]

**Pull request merged**